presto-native-execution/presto_cpp/main/PrestoTask.cpp

```diff
@@ -590,10 +590,20 @@ void PrestoTask::updateExecutionInfoLocked(
   prestoTaskStats.outputPositions = 0;
   prestoTaskStats.outputDataSizeInBytes = 0;
 
-  prestoTaskStats.queuedDrivers = veloxTaskStats.numQueuedDrivers;
-  prestoTaskStats.totalDrivers = veloxTaskStats.numTotalDrivers;
+  // Presto Java reports number of drivers to number of splits in Presto UI
+  // because split and driver are 1 to 1 mapping relationship. This is not true
+  // in Prestissimo where 1 driver handles many splits. In order to quickly
+  // unblock developers from viewing the correct progress of splits in
+  // Prestissimo's coordinator UI, we put number of splits in total, queued, and
+  // finished to indicate the progress of the query. Number of running drivers
+  // are passed as it is to have a proper running drivers count in UI.
+  //
+  // TODO: We should really extend the API (protocol::TaskStats and Presto
```

Comment:
This is quite hacky and will be confusing for Prestissimo engineers who don't come from Presto Java.

We should ideally extend protocol::TaskStats and extend the API. How expensive is this ? At the very least can we create a Jira issue ? I'll get it triaged to someone in our team if you'll don't have the cycles.

@tdcmeehan @amitkdutta

Reply:
This was the situation from the start of Prestissimo and we made a regression during a fix (#23149).  Just put it back to what it was.

Reply:
Ah I see. Do you want to create a github issue for the protocol::TaskStats cleanup ? There are engineers on our team who want to fix that issue.

Reply:
Sure #23758

Reply:
Thanks